VMBackup/main/handle.py

```diff
@@ -165,33 +164,41 @@ def snapshot():
         errMsg = 'Failed to do the snapshot with error: %s, stack trace: %s' % (str(e), traceback.format_exc())
         backup_logger.log(errMsg, False, 'Error')
     snapshot_done = True
-
-
-def freeze_watcher():
-    global backup_logger,run_status,error_msg,snapshot_done
-    #wait 5 minutes before the snapshoting.
-    try:
-        for i in range(0, 50):
-            if(snapshot_done):
-                backup_logger.log('T:W snapshot is done', False)
-                break;
-            sleep(6)
-        if not snapshot_done:
-            run_result = CommonVariables.error
-            run_status = 'error'
-            error_msg = 'T:W Snapshot timeout'
-            backup_logger.log(error_msg, False, 'Warning')
-    except Exception as e:
-        errMsg = 'Failed to do  the snapshot because of exception in freeze watcher'
-        backup_logger.log(errMsg, False, 'Error')
 def daemon():
-    global MyPatching,backup_logger,hutil,run_result,run_status,error_msg,freezer,para_parser
+    global MyPatching,backup_logger,hutil,run_result,run_status,error_msg,freezer,para_parser,snapshot_done
     #this is using the most recent file timestamp.
     hutil.do_parse_context('Executing')
     freezer = FsFreezer(patching= MyPatching, logger = backup_logger)
     global_error_result = None
     # precheck
     freeze_called = False
+
+    configfile='/var/lib/waagent/extension_config/config.ini'
+    thread_timeout=str(60)
+    if not os.path.exists(configfile):
+        directory_present= True
+        try:
+            os.makedirs(os.path.dirname(configfile))
+        except Exception as e:
+            if e.errno != err.EEXIST:
+                directory_present=False
+        if(directory_present):
+            try:
+                fopen=open(configfile,'w')
+                fopen.write('[SnapshotThread]\ntimeout=60')
+                fopen.close()
+            except Exception as e:
+                errMsg='Config file not present and cannot be created'
+                backup_logger.log(errMsg, False, 'Warning')
+    try:
+        config = ConfigParser.ConfigParser()
+        config.read(configfile)
+        thread_timeout= config.get('SnapshotThread','timeout')
```

Comment:
ok fair point, there's nothing wrong with the existing code in terms of functionality, it is just less maintainable.I think we can merge if you just commit the config file path change.